akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientServerSpec.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import akka.http.impl.util.{ AkkaSpecWithMaterializer, ExampleHttpContexts }
+import akka.http.scaladsl.model.http2.{ Http2StreamIdHeader, RequestResponseAssociation }
+import akka.http.scaladsl.model.{ AttributeKey, HttpMethods, HttpRequest, HttpResponse }
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.http.scaladsl.{ Http, Http2 }
+import akka.stream.scaladsl.{ Sink, Source }
+import akka.stream.testkit.{ TestPublisher, TestSubscriber }
+import akka.testkit.TestProbe
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.{ Future, Promise }
+
+class Http2ClientServerSpec extends AkkaSpecWithMaterializer(
+  """akka.http.server.remote-address-header = on
+     akka.http.server.http2.log-frames = on
+     akka.http.server.log-unencrypted-network-bytes = 100
+     akka.http.server.preview.enable-http2 = on
+     akka.actor.serialize-messages = false
+  """) with ScalaFutures {
+  case class RequestId(id: String) extends RequestResponseAssociation
+  val requestIdAttr = AttributeKey[RequestId]("requestId")
```

Comment:
Here's how you use it: you create a token type and extend from `RequestResponseAssociation`. It can be any type.